pkg/kubelet/kubelet_pods.go

```diff
@@ -387,8 +387,8 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 	// project the services in namespace ns onto the master services
 	for i := range services {
 		service := services[i]
-		// ignore services where ClusterIP is "None" or empty
-		if !v1helper.IsServiceIPSet(service) {
+		// ignore services where ClusterIP is "None" or empty but always add the master service
+		if !v1helper.IsServiceIPSet(service) && !(service.Namespace == kl.masterServiceNamespace && service.Name == masterServiceName) {
```

Comment:
Yes, in case someone activates the feature and still has old Kubelets running. 
How should this prevented? Documentation?